src/abstractops.cpp

```diff
@@ -1100,7 +1104,8 @@ Expr AbsFpEncoding::truncate(const smt::Expr &f, aop::AbsFpEncoding &tgt) {
   return Expr::mkIte(isnan(f), tgt.nan(),
           Expr::mkIte(f == infinity(), tgt.infinity(),
           Expr::mkIte(f == infinity(true), tgt.infinity(true),
-          Expr::mkIte(limit_bits != limit_zero,
+          Expr::mkIte(limit_bits != limit_zero |
+                      floored_value.uge(tgt.getMagnitudeBits(tgt.infinity())),
```

Comment:
Could you factor out `floored_value.uge(tgt.getMagnitudeBits(tgt.infinity())` as a separate variable please?